src/core/Akka.Persistence/Serialization/PersistenceMessageSerializer.cs

```diff
@@ -56,19 +58,18 @@ private PersistentMessage GetPersistentMessage(IPersistentRepresentation persist
         private PersistentPayload GetPersistentPayload(object obj)
         {
             Serializer serializer = system.Serialization.FindSerializerFor(obj);
-            PersistentPayload payload = new PersistentPayload();
+            var payload = new PersistentPayload();
 
-            if (serializer is SerializerWithStringManifest)
+            if (serializer is SerializerWithStringManifest serializer2)
             {
-                string manifest = ((SerializerWithStringManifest)serializer).Manifest(obj);
+                string manifest = serializer2.Manifest(obj);
                 payload.PayloadManifest = ByteString.CopyFromUtf8(manifest);
             }
             else
             {
                 if (serializer.IncludeManifest)
                 {
-                    var payloadType = obj.GetType();
-                    payload.PayloadManifest = ByteString.CopyFromUtf8(payloadType.AssemblyQualifiedName);
+                    payload.PayloadManifest = ByteString.CopyFromUtf8(obj.GetType().TypeQualifiedName());
```

Comment:
Type qualified name is without the assemblyname right ?

Reply:
It looks like this is an extension method in the Akka.Util package, which will attempt to serialize without the assembly name if it is part of the core assembly.  I'm a little bit nervous on how well this works across runtime versions and if there are going to be an issues during upgrades.  The change can be found at https://github.com/akkadotnet/akka.net/commit/d3328cab47e471783f8c7a4b3379188a5c9ac02a

Reply:
Yeah, so I just did a quick comparison.  A mono 4.8 app has a core assembly name of mscorlib, while a .net core standard app has a core assembly name of System.Private.CoreLib.  This could cause issues with people that are trying to switch runtimes.  I'll create an issue.

Reply:
@joshgarnett is this an issue with this specific change, or also with the previous code?

Reply:
Actually I take that back, in theory the default should always deserialize properly for types with no assembly included?

Reply:
@joshgarnett Have you found some bugs? Because I've covered this scenario https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka/Util/TypeExtensions.cs#L63

Reply:
Just so I have it straight - the issue here is when someone previously was serializing messages with a mono 4.8 app, then switch to a .net core app, they wouldn't be able to deserialize the messages based on the type specified in the manifest, because it will not be `System.String, System.Private.CoreLib`, but instead be `System.String, mscorlib`

Reply:
@joshgarnett, also related to this block of code - what if the serializer is not set to `serializer.IncludeManifest = true`?  How would we be able to `ReadEvent`?

Reply:
@heynickc Anyway, we have used the similar method in 1.2.x

Reply:
Yeah, I think we are fine, I retracted my comment right after yours @heynickc I was concerned at first, but all this is doing is grouping together objects that would be under the core assembly, which should be the same set of objects regardless of the actual name of the core assembly.

Reply:
@alexvaluyskiy ok sorry, I was in the wrong vacinity.  this is where I was curious about such an issue: https://github.com/akkadotnet/akka.net/blob/dev/src/contrib/persistence/Akka.Persistence.Sql.Common/Journal/QueryExecutor.cs#L656

Reply:
@heynickc as for IncludeManifest, that is how serializers should work, right?  If they do not set IncludeManifest to true, then the assumption is that the serializer does not need the manifest in order to deserialize the data properly.

Reply:
@joshgarnett gotcha, thanks.  I ran into the issue testing backwards compatibility, but I may have been testing an edge case that wasn't possible.